### PR TITLE
feat(request-manager): intercept WebSocket handshake

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description
 intercept WebSocket handshake and create random identity for each WebSocket connection

## Related Issue

Prerequisite for https://github.com/trezor/trezor-suite/issues/7661

### CODE REVIEW

File: packages/request-manager/e2e/interceptor.test.ts [hunk 1]
```diff
@@ -145,6 +145,53 @@ describe('Interceptor', () => {
         });
     });
 
+    describe('WebSocket', () => {
+        // 'ws' package is not a part of this project
```
Comment (karliatto) [inline packages/request-manager/e2e/interceptor.test.ts]: If we use it here, I guess it is now part of the project. Why not adding it as dev dependency?

**Review by karliatto — COMMENTED** [review]
Very nice and clear!
I have tested it and did some network analysis to check it works as expected and it does. 

I think now we should get rid of agent setting in blockchain link `https://github.com/trezor/trezor-suite/blob/feat/intercept-ws-handshake/packages/blockchain-link/src/workers/blockbook/index.ts#L373`

And maybe the setting of  setProxy in connect from `https://github.com/trezor/trezor-suite/blob/feat/intercept-ws-handshake/packages/suite-desktop/src/modules/tor.ts#L144`. Or is there any other reason for setProxy in connect? 

We can do this in this PR or we can create a new issue as you wish @szymonlesisz

**Review by karliatto — APPROVED** [review]

**Comment by szymonlesisz** [comment]
@karliatto i think setting agent in blockchain-link could stay as it is, it still can be used as a part of TrezorConnect api.

however `suite-desktop/src/modules/tor.ts#L144` should be modified, suite doesn't need `TrezorConnect.setProxy` method anymore, but keep in mind that `TrezorConnect.setProxy` is also responsible for all backends disconnection, so it's not only about removal

File: packages/request-manager/e2e/interceptor.test.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/request-manager/e2e/interceptor.test.ts]: sure, why not.
rebased and forcepushed with "ws" dependency

**Comment by karliatto** [comment]
> `TrezorConnect.setProxy` is also responsible for all backends disconnection

@szymonlesisz what does it mean? I am not sure I understand that part.

**Comment by szymonlesisz** [comment]
i was thinking reconnection and wrote disconnection :)

currently it work like this:
```
modules/tor.ts > TrezorConnect.setProxy > [inside connect]: disconnect all running backends and recconnect them with proxy
```

now, if we just remove that part (TrezorConnect.setProxy) then websocket will never be disconnected, therefore agent will never be assigned by request-manager

**Comment by karliatto** [comment]
> now, if we just remove that part (TrezorConnect.setProxy) then websocket will never be disconnected, therefore agent will never be assigned by request-manager

Ok, I undestand what you mean. So we need either a new connect method `reconnect` or just use `setProxy` that does the same.

@szymonlesisz What do you think? I have the feeling that having new method `reconnect` is a better idea.

**Comment by szymonlesisz** [comment]
well, current usage of setProxy does not break anything, agent set by blockchain-link will be overriden by request-manager. we can keep it as and do a followup issue (i dont have a time to deal with it right now) to create `TrezorConnect.blockchainConnect({ coin })` + `TrezorConnect.blockchainReconnect()` and use it with combination of `TrezorConnect.blockchainDisconnect()`

additionally `TrezorConnect.blockchainConnect` could be used [here](https://github.com/trezor/trezor-suite/blob/develop/suite-common/wallet-core/src/blockchain/blockchainThunks.ts#L152)

**Comment by karliatto** [comment]
Ok, I created issue https://github.com/trezor/trezor-suite/issues/7689 to follow it up.

**Pull request merged**